src/app/templates/partners-promotion/index.tsx

```diff
@@ -89,25 +101,47 @@ export const PartnersPromotion = memo<PartnersPromotionProps>(({ variant, id, pa
 
   return (
     <div className={clsx('w-full relative', !adIsReady && (isImageAd ? styles.imageAdLoading : styles.textAdLoading))}>
-      {shouldUseOptimalAd ? (
-        <OptimalPromotion
-          variant={variant}
-          isVisible={adIsReady}
-          onAdRectSeen={handleAdRectSeen}
-          onClose={handleClosePartnersPromoClick}
-          onReady={handleAdReady}
-          onError={handleOptimalError}
-        />
-      ) : (
-        <HypelabPromotion
-          variant={variant}
-          isVisible={adIsReady}
-          onAdRectSeen={handleAdRectSeen}
-          onClose={handleClosePartnersPromoClick}
-          onReady={handleAdReady}
-          onError={handleHypelabError}
-        />
-      )}
+      {(() => {
+        switch (providerName) {
+          case 'Optimal':
+            return (
+              <OptimalPromotion
+                variant={variant}
+                isVisible={adIsReady}
+                pageName={pageName}
+                onAdRectSeen={handleAdRectSeen}
+                onClose={handleClosePartnersPromoClick}
+                onReady={handleAdReady}
+                onError={handleOptimalError}
+              />
+            );
+          case 'HypeLab':
+            return (
+              <HypelabPromotion
+                variant={variant}
+                isVisible={adIsReady}
+                pageName={pageName}
+                onAdRectSeen={handleAdRectSeen}
+                onClose={handleClosePartnersPromoClick}
+                onReady={handleAdReady}
+                onError={handleHypelabError}
+              />
+            );
+          case 'Persona':
+            return (
+              <PersonaPromotion
+                id={id}
+                isVisible={adIsReady}
+                pageName={pageName}
+                onAdRectSeen={handleAdRectSeen}
+                onClose={handleClosePartnersPromoClick}
+                onReady={handleAdReady}
+                onError={handlePersonaError}
+              />
+            );
+        }
+      })()}
```

Comment:
```suggestion
      {providerName === 'Optimal' && (
        <OptimalPromotion
          variant={variant}
          isVisible={adIsReady}
          pageName={pageName}
          onAdRectSeen={handleAdRectSeen}
          onClose={handleClosePartnersPromoClick}
          onReady={handleAdReady}
          onError={handleOptimalError}
        />
      )}
      {providerName === 'HypeLab' && (
        <HypelabPromotion
          variant={variant}
          isVisible={adIsReady}
          pageName={pageName}
          onAdRectSeen={handleAdRectSeen}
          onClose={handleClosePartnersPromoClick}
          onReady={handleAdReady}
          onError={handleHypelabError}
        />
      )}
      {providerName === 'Persona' && (
        <PersonaPromotion
          id={id}
          isVisible={adIsReady}
          pageName={pageName}
          onAdRectSeen={handleAdRectSeen}
          onClose={handleClosePartnersPromoClick}
          onReady={handleAdReady}
          onError={handlePersonaError}
        />
      )}
```
Creating a function on each render may cause memory overconsumption. Please check the changes with linter before applying